web/themes/contrib/civictheme/includes/views.inc

```diff
@@ -250,14 +255,150 @@ function _civictheme_preprocess_views__exposed_form__group_filter(array &$variab
+          $temp_query_params = $query_params;
+          unset($temp_query_params[$key][$value_key]);
+          if (empty($temp_query_params[$key])) {
+            unset($temp_query_params[$key]);
+          }
+          $filter_url = _civictheme_preprocess_views_view__selected_filters_list_create_filter_url($base_url, $temp_query_params, $key, $item, $filter_types);
+          $selected_filters[$key . '_' . $item . '_' . $value_key] = $filter_url;
+        }
+      }
+    }
+  }
+  return $selected_filters;
+}
+
+/**
+ * Create the URL for the selected filter.
+ */
+function _civictheme_preprocess_views_view__selected_filters_list_create_filter_url(string $base_url, array $query_params, string $key, string|int $value, array $filter_types): array {
+  $query = http_build_query($query_params);
+  return [
+    'url' => !empty($query) ? ($base_url . '?' . $query) : $base_url,
+    'text' => _civictheme_preprocess_views_view__selected_filters_list__create_filter_label($key, $value, $filter_types),
+  ];
+}
+
+/**
+ * Get the filter types from the view.
+ *
+ * @param \Drupal\views\ViewExecutable $view
+ *   The view executable object.
+ *
+ * @return array
+ *   Array of exposed filters with their properties:
+ *   - name: The human readable label
+ *   - type: The filter plugin ID
+ *   - machine_name: The filter ID
+ */
+function _civictheme_preprocess_views_view__selected_filters_get_filter_types(ViewExecutable $view): array {
+  $exposed_filters = [];
+  $filters = $view->display_handler->getHandlers('filter');
+  foreach ($filters as $filter_id => $filter) {
+    if ($filter instanceof FilterPluginBase && $filter->isExposed()) {
+      $exposed_filters[$filter->options['expose']['identifier']] = [
+        'name' => $filter->options['expose']['label'],
+        'type' => $filter->getPluginId(),
+        'machine_name' => $filter_id,
+      ];
+    }
+  }
+  return $exposed_filters;
+}
+
+/**
+ * Create the label correctly for each of the selected filters.
+ *
+ * @param string $key
+ *   The filter key.
+ * @param string|int $value
+ *   The filter value.
+ * @param array $filter_types
+ *   Array of filter types containing metadata about each filter.
+ *
+ * @return string
+ *   The formatted filter label in the format "Key: Value".
+ */
+function _civictheme_preprocess_views_view__selected_filters_list__create_filter_label(string $key, string|int $value, array $filter_types): string {
+  switch ($filter_types[$key]['type']) {
+    case 'bundle':
+      $node_type = \Drupal::entityTypeManager()->getStorage('node_type')->load($value);
+      $value_label = $node_type ? $node_type->label() : $value;
+      break;
+
+    case 'taxonomy_index_tid':
+      $taxonomy_term = \Drupal::entityTypeManager()->getStorage('taxonomy_term')->load($value);
+      $value_label = $taxonomy_term ? $taxonomy_term->label() : $value;
+      break;
+
+    default:
+      $value_label = $value;
+  }
+  return $filter_types[$key]['name'] . ': ' . $value_label;
```

Comment:
this function should return only a label and the formatting of this label should take place outside of this function